### ISSUE TITLE
[RFC] Can We Show Component Labels in the UI?

### ISSUE THREAD

**Issue by ProticM** [backstage/backstage#7653]
**Status:** Open for comments

## Need

I need to show the labels of the `Component` kind in the Backstage portal. Are they shown somewhere already?

## Proposal

Show them in the catalog table and the details page if possible.

In the catalog table, we can create a custom column, but it seems that there is a problem with the search. After adding a new column that column is not taken into account when searching.

Is it possible to show the label on the component details page?

**Comment by ProticM** [comment]
Hi @Crevil 

I was wondering if it is possible for us to show the `labels` (https://backstage.io/docs/features/software-catalog/descriptor-format#labels-optional) in the UI?

For example, if we attach a tag to the component metadata (YAML file) it is visible in the catalog table (home page) and the component details page. I wanted to know if there is a place where the labels from the YAML are displayed?

I did not find such a place and I experimented with adding a custom column in the catalog table like explained here https://github.com/backstage/backstage/issues/7004

**Comment by ProticM** [comment]
@Rugvip 

The field is pretty important to be visible (if possible, at the same level as the owner, or the system of the component for example) therefore having it behind some popup or similar would not be our primary solution

> Would also say that this might be a case where you want to extend the catalog model to include your own custom fields

Yes, this is what I proposed as a possible solution in my internal team instead of having tags for those values.

To me, for these fields (we have, for example, a launch date that we want to show in the UI), it makes more sense to be a custom field in even an additional column in the grid instead of them being listed as a tag (on a side that it will have a weird in that case 😄)

**Comment by ProticM** [comment]
Hi @Rugvip  If there is no backstage specific answer to this:

> If not, is there a recommended way on how to add a card to the details page for example?

We can close this one.

I'm asking because there is recommended approach how to create columns explained here https://github.com/backstage/backstage/issues/7004

**Comment by Rugvip** [comment]
@ProticM If you mean customizing the About card then there's no pattern for that atm. Adding cards to the overview pages you can do though, just modify your own `packages/app/src/components/catalog/EntityPage.tsx` to include the new cards somewhere in the grid that's typically defined as `overviewContent`. The card itself you'll probably want to put in a small plugin that exports it

**Comment by ProticM** [comment]
@Rugvip Yes, thanks. This is how I did it, we have a plugin for custom cards and they are showing without issues
